ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,430 @@
+                        h5_node = file.root[node]
+                        data = self.drop_column(h5_node[:], "tels_with_trigger")
+                        self.output_file.create_table(
+                            group_path,
+                            table_name,
+                            filters=h5_node.filters,
+                            obj=data,
+                        )
+
+                    else:
+                        file.copy_node(node, newparent=target_group)
+
+    def _create_group(self, node):
+        head, tail = os.path.split(node)
+        self.output_file.create_group(head, tail, createparents=True)
+
+    @staticmethod
+    def drop_column(array, column):
+        from numpy.lib.recfunctions import repack_fields
+
+        cols = list(array.dtype.names)
+        if column in cols:
+            cols.remove(column)
+
+        return repack_fields(array[cols])
+
+    def start(self):
+        merged_files_counter = 0
+
+        for i, current_file in enumerate(
+            tqdm(
+                self.input_files,
+                desc="Merging",
+                unit="Files",
+                disable=not self.progress_bar,
+            )
+        ):
+
+            with tables.open_file(current_file, mode="r") as file:
+                if i == 0:
+                    # Check if first file is simulation
+                    if "/simulation" not in file.root:
+                        self.usable_nodes = self.usable_nodes - simu_nodes
+                        self.log.info("Merging real data")
+                    else:
+                        self.log.info("Merging simulation-files")
+
+                if self.check_file_broken(file) is True:
+                    if self.skip_broken_files is True:
+                        continue
+                    else:
+                        self.log.critical("Broken file detected.")
+                        sys.exit(1)
+
+                self.merge_tables(file)
+                self.add_image_statistics(file)
+
+            PROV.add_input_file(str(current_file))
+            merged_files_counter += 1
+
+        self.log.info(
+            f"{merged_files_counter} out of {len(self.input_files)} files "
+            "has been merged!"
+        )
+
+    def finish(self):
+        activity = PROV.current_activity.provenance
+        DL1_DATA_MODEL_VERSION = "v1.0.0"
+
+        reference = meta.Reference(
+            contact=meta.Contact(name="", email="", organization="CTA Consortium"),
+            product=meta.Product(
+                description="Merged DL1 Data Product",
+                data_category="S",
+                data_level="DL1",
+                data_association="Subarray",
+                data_model_name="ASWG DL1",
+                data_model_version=DL1_DATA_MODEL_VERSION,
+                data_model_url="",
+                format="hdf5",
+            ),
+            process=meta.Process(type_="Simulation", subtype="", id_=0),
```

Comment:
So perhaps for now, only set it to  "simulation" if the `is_simulation` flag is set to True, otherwise it should be "observation".  

For the `id` field, right now it is not defined in the data model doc if this is a merged data file, so I guess 0 is fine, or we could set it to the PROD id (e.g. "Prod3"), but right now I guess there is no way to determine that from the simulation files.   This will be better defined in the next revision of the data model...

Reply:
This goes back to the comments I made in the stage1 tool. Shouldn't we expose parts of these metadata fields via the config system, so that users can set things like the contact, email, and prod from the config file or cli?